plugins/source/azure/client/client.go

```diff
@@ -363,3 +367,31 @@ func (c *Client) withBillingPeriod(billingPeriod *armbilling.Period) *Client {
 	newC.BillingPeriod = billingPeriod
 	return &newC
 }
+
+func (c *Client) GetStorageAccountKey(ctx context.Context, acc *armstorage.Account) (string, error) {
+	if val, ok := c.storageAccountKeys.Load(*acc.Name); ok {
```

Comment:
If this function happens to get called for the same storage account in parallel by 100 goroutines, we could still end up making 100 requests, because there's some delay between this line and the `Store` call lower down.

If we want to avoid that potential, we should have some locking mechanism around this. Maybe a `sync.Map` with  that indicates whether an storage account is already in the process of being loaded?

There's probably a better way of doing this, but I'm thinking something along these lines:

```
	mut := c.storageAccountLoading.LoadOrStore(*acc.Name, &sync.Mutex{})
	mut.Lock()
	defer mut.Unlock()
	if val, ok := c.storageAccountKeys.Load(*acc.Name); ok {
		return val.(string), nil
	}
```

Reply:
there's also sync.Once to be considered for init once (per sync)